### DISCUSSION TITLE
xaringan slides infinite moon reader

### DISCUSSION THREAD

**Discussion by liutiming** [ropensci/targets#307] [General]
Hi @wlandau, as you also use Xaringan slides, have you tried to render Xaringan with vscode on a cluster?

I am not sure what is the best way to work with xaringan and targets - the infinite moon reader function does not seem to support targets...

**Comment by wlandau** [comment]
I routinely render xaringan slide decks as downstream targets. The technique  is the same as ordinary dependency-aware R Markdown: https://books.ropensci.org/targets/files.html#literate-programming. What seems to be the issue in your case?

**Reply by liutiming** [reply]
Mostly adjustment of layouts: image size, font size etc. so that 1. they are large enough 2. they all fit in the same slide

I find infinite_moon_reader to help with local deployment but can't figure out how to use it with targets...

**Reply by wlandau** [reply]
I usually set deployment = “main” for slides so they render locally. Since I am just loading upstream targets, rendering does not take long.

**Reply by liutiming** [reply]
It is the cognitive load of task switching I think. It will usually take about 30 sec - 1 min to load and during that time I will often do something else, and then I will come back to look at the slide. So it is difficult to engage in "deep work" alongside making slides.

**Reply by wlandau** [reply]
Do regular knitr layout/format options stop working with tar_render(), even when you set these options in the yaml front matter? Sorry, I still do not understand the connection with targets specifically.

**Reply by wlandau** [reply]
Maybe create lighter downstream summary targets for the report to make rendering faster?

**Reply by liutiming** [reply]
Oh no. I believe the layout is still the same. I think it is just that the process to constantly tar_make() instead of seeing the changes instantly with infinite_moon_reader is a bit taxing.

**Reply by liutiming** [reply]
Yup. Now I think an alternative is perhaps to use RStudio Server and put the report to be rendered in the root directory so that `_targets/` is accessible. Then use `infinite_moon_reader` there. After one is satisfied with the layout, then they can use `tar_render` to render the report. The downside is that one has to source and load the scripts and libraries in the Rmd file as well.

I have not tried this proposal yet, because it is tedious to set up RStudio Server on my cluster (need to work with openstack and other technologies to get it running)

**Reply by liutiming** [reply]
and I just find vscode more light-weight, intuitive, and extensible to work with than Rstudio...

**Reply by wlandau** [reply]
If you are iterating a lot on small changes, then yes, I recommend running most iterations through the IDE of your choice. Is the issue rendering through VSCode specifically? I thought there was an extension that made it just as easy as the Knit button in RStudio.

I tried to switch to VSCode earlier this year, but it did not stick because auto-indent does not work for R, and AFAIK VSCode does not have multi-line cursor support.

**Reply by liutiming** [reply]
I thought the multi-line cursor support in vscode was quite good... have you tried ctrl + alt + up/down 
or 
mouse middle click?

**Reply by wlandau** [reply]
Oh interesting, I did not know that. The keyboard shortcuts do not work for me though. The middle click does, but that is the wheel on my mouse, so my finger is likely to slip.
